library/src/main/java/com/novoda/downloadmanager/lib/RequestBatch.java

```diff
@@ -0,0 +1,50 @@
+package com.novoda.downloadmanager.lib;
+
+import android.content.ContentValues;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RequestBatch {
+
+    private final BatchInfo batchInfo;
+    private final List<Request> requests;
+
+    public static RequestBatch newInstance(String title, String description, String bigPictureUrl) {
+        BatchInfo batchInfo = new BatchInfo(title, description, bigPictureUrl);
+        return new RequestBatch(batchInfo, new ArrayList<Request>());
+    }
+
+    RequestBatch(BatchInfo batchInfo, List<Request> requests) {
+        this.batchInfo = batchInfo;
+        this.requests = requests;
+    }
+
+    public String getTitle() {
+        return batchInfo.getTitle();
+    }
+
+    public String getDescription() {
+        return batchInfo.getDescription();
+    }
+
+    public String getBigPictureUrl() {
+        return batchInfo.getBigPictureUrl();
+    }
+
+    public List<Request> getRequests() {
```

Comment:
It would be awesome if this could be avoided so we don't expose the internal representation of the data that this class holds, that would problably mean to push logic into this class :)